gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/spanner/SpannerOptionsStubClassComposer.java

```diff
@@ -0,0 +1,376 @@
+import com.google.api.generator.gapic.model.Service;
+import com.google.api.generator.gapic.utils.JavaStyle;
+import com.google.api.pathtemplate.PathTemplate;
+import com.google.api.pathtemplate.ValidationException;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import io.grpc.spanner.SpannerOptionsConfig;
+import io.grpc.spanner.SpannerOptionsConfig.OptionMapperConfig;
+import io.grpc.spanner.SpannerOptionsConfig.OptionMapperConfigList;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Generated;
+
+
+/**
+ * Generate an Option class which represents an Option parameter.
+ */
+public class SpannerOptionsStubClassComposer implements ClassComposer {
+  private static final SpannerOptionsStubClassComposer INSTANCE =
+      new SpannerOptionsStubClassComposer();
+  private static final TypeStore FIXED_TYPESTORE = createStaticTypes();
+
+  private static final VaporReference READ_AND_QUERY_OPTION =
+      createVaporReference("com.google.cloud.spanner", "ReadAndQueryOption");
+  private static final VaporReference TRANSACTION_OPTION =
+      createVaporReference("com.google.cloud.spanner", "TransactionOption");
+  private static final VaporReference INTERNAL_OPTION_REF =
+      createVaporReference("com.google.cloud.spanner", "InternalOption");
+  private static final VaporReference OPTIONS =
+      createVaporReference("com.google.cloud.spanner", "Options");
+  private static final Variable OPTIONS_VAR = createVarFromVaporRef(OPTIONS, "options");
+
+  public static SpannerOptionsStubClassComposer instance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public GapicClass generate(GapicContext context, Service service) {
+
+    // Read the configuration for mapper
+    SpannerOptionsConfig.OptionMapperConfigList optionMapperConfigList = parseSpannerOptionsConfig();
+
+    // from the proto, parse the applicable fields which need to be autogenerated
+    Map<Field, String> fields = getApplicableProtoFields(context, optionMapperConfigList);
+
+    List<Statement> memberVariables = new ArrayList<>();
+    List<ClassDefinition> optionInnerClasses = new ArrayList<>();
+    List<MethodDefinition> setterMethods = new ArrayList<>();
+    List<MethodDefinition> hasMethods = new ArrayList<>();
+    List<MethodDefinition> getterMethods = new ArrayList<>();
+    List<MethodDefinition> allMethods = new LinkedList<>();
+
+    for(Map.Entry<Field, String> field: fields.entrySet()) {
+
+      // generate public method to set option value
+      /**
+       * public static Options.TransactionOption maxBatchingDelayMs(int maxBatchingDelayMs) {
```

Comment:
@blakeli0 There is an error over here due to a mismatch between the desired return type and the return expression. Refer - https://github.com/googleapis/sdk-platform-java/issues/2122